XMonad/Layout/IndependentScreens.hs

```diff
@@ -172,17 +195,33 @@ whenCurrentOn s pp = pp
         _ -> ppOutput pp out
     }
 
--- | If @vSort@ is a function that sorts 'WindowSpace's with virtual names, then @marshallSort s vSort@ is a function which sorts 'WindowSpace's with physical names in an analogous way -- but keeps only the spaces on screen @s@.
-marshallSort :: ScreenId -> ([WindowSpace] -> [WindowSpace]) -> ([WindowSpace] -> [WindowSpace])
+-- | Filter workspaces that are on current screen.
+workspacesOn :: ScreenId -> [PhysicalWindowSpace] -> [PhysicalWindowSpace]
+workspacesOn s = filter (\ws -> unmarshallS (W.tag ws) == s)
+
+-- | @vSort@ is a function that sorts 'VirtualWindowSpace's with virtual names.
+-- @marshallSort s vSort@ is a function which sorts 'PhysicalWindowSpace's with virtual names, 
+-- but keeps only the 'WindowSpace'\'s on screen @s@.
+marshallSort :: ScreenId -> ([VirtualWindowSpace] -> [VirtualWindowSpace]) -> ([PhysicalWindowSpace] -> [PhysicalWindowSpace])
 marshallSort s vSort = pScreens . vSort . vScreens where
-    onScreen ws = unmarshallS (tag ws) == s
-    vScreens    = map unmarshallWindowSpace . filter onScreen
+    vScreens    = map unmarshallWindowSpace . workspacesOn s
     pScreens    = map (marshallWindowSpace s)
+-- This marshallSort has a bug: see https://github.com/xmonad/xmonad-contrib/issues/420
+-- To use 'getSortByIndex', please use the better bug-fixed solution provided below. 
+
+-- This marshallSort is the better solution.
+-- It sorts directly the physical workspace names on the current screen.
+-- > marshallSort :: ScreenId -> ([PhysicalWindowSpace] -> [PhysicalWindowSpace]) -> ([PhysicalWindowSpace] -> [PhysicalWindowSpace])
+-- > marshallSort s vSort = vSort . workspacesOn s
```

Comment:
I think we don't need to mention this as it's the default in `marshallPP`. Just add an example how to use `marshallSort` to provide a custom virtual names sort on top of `marshallPP`, and put that example in the haddock comment so that it's visible in the generated documentation.

Reply:
I have updated with an example, not sure if it is appropriate.